Sources/xcproj/PBXProjObjects+Helpers.swift

```diff
@@ -15,7 +16,160 @@ public extension PBXProj.Objects {
+                    path: options.contains(.withoutFolder) ? nil : groupName,
+                    usesTabs: usesTabs,
+                    indentWidth: indentWidth,
+                    tabWidth: tabWidth
+                )
+                reference = generateReference(newGroup, groupName)
+                addObject(newGroup, reference: reference)
+                toGroup.children.append(reference)
+            }
+            toGroup = newGroup
+        }
+        return (reference, newGroup)
+    }
+
+    /// Adds file at the give path to the project. Returns new file and its reference.
+    ///
+    /// - Parameters:
+    ///   - filePath: path to the file
+    ///   - sourceTree: file source tree, default is `.group`.
+    ///   - name: file name, by default gets file name from the path
+    /// - Returns: new file and its reference.
+    public func addFile(at filePath: Path,
+        sourceTree: PBXSourceTree = .group,
+        name: String? = nil,
+        fileEncoding: Int? = nil,
+        explicitFileType: String? = nil,
+        lastKnownFileType: String? = nil,
+        includeInIndex: Int? = nil,
+        usesTabs: Int? = nil,
+        lineEnding: Int? = nil,
+        xcLanguageSpecificationIdentifier: String? = nil
+        ) -> (String, PBXFileReference) {
+
+        if let existingFileReference = fileReferences.first(where: { $0.value.path == filePath.string }) {
+            return existingFileReference
+        }
+
+        let fileReference = PBXFileReference(
+            sourceTree: sourceTree,
+            name: name ?? filePath.lastComponent,
+            fileEncoding: fileEncoding,
+            explicitFileType: explicitFileType ?? PBXFileReference.fileType(path: filePath),
+            lastKnownFileType: lastKnownFileType ?? PBXFileReference.fileType(path: filePath),
+            path: filePath.string,
+            includeInIndex: includeInIndex,
+            usesTabs: usesTabs,
+            xcLanguageSpecificationIdentifier: xcLanguageSpecificationIdentifier
+        )
+        let reference = generateReference(fileReference, filePath.string)
+        addObject(fileReference, reference: reference)
+        return (reference, fileReference)
+    }
+
+    /// Adds file to the given group.
+    /// If group already contains file with given reference method does nothing.
+    ///
+    /// - Parameter group: group to add file to
+    /// - Parameter reference: file reference
+    public func addFile(toGroup group: PBXGroup, reference: String) {
+        guard !group.children.contains(reference) else { return }
+        group.children.append(reference)
+    }
+
+    /// Adds file to the given target's sources build phase
+    /// If target's sources build phase can't be found or it already contains file with given reference method does nothing.
+    ///
+    /// - Parameter target: target obejct
+    /// - Parameter reference: file reference
+    public func addBuildFile(toTarget target: PBXTarget, reference: String) {
```

Comment:
What do you think about returning the `PBXBuildFile` here?

Reply:
I'll return file and reference to match other APIs